### ISSUE TITLE
Confusing error when `theme(legend.direction = .)` incorrectly entered

### ISSUE THREAD

**Issue by ngreifer** [tidyverse/ggplot2#4930]
When running `ggplot()` with a legend and using `theme(legend.position = .)` to change the legend direction, an invalid argument causes an unclear error. I also feel like abbreviations should be allowed for this argument since it only accepts two strings.

```
library(ggplot2)
ggplot(mpg, aes(x = cty, fill = drv)) +
  geom_histogram() +
  theme(legend.direction = "v")

## Error in if (!g$title.position %in% c("top", "bottom", "left", "right")) { : 
##  argument is of length zero
```

**Comment by teunbrand** [comment]
Thanks, that is indeed an inelegant error message. After #4879 gets merged, this will return the following error message:

> Error:
! `direction` must be one of "horizontal" or "vertical", not "v".
ℹ Did you mean "vertical"?

Quoting from the tidyverse style [guide](https://design.tidyverse.org/def-enum.html#how-do-i-use-it):

> I generally believe that partial matching is a bad idea, because it makes code harder to read.

So I don't think there is a good chance that ggplot2 will allow this as a partially matched argument.

**Comment by teunbrand** [comment]
I just realised that this is a duplicate of #4364, so I'll close this issue in favour of the other one.
